content/how-to/bpmn/trigger-workflow.md

```diff
@@ -0,0 +1,145 @@
+
+The `customResponse` is a special variable to return data in the response to clients that run `createAndRunBPMNSync` operations.
+
+You can set the `customResponse` in any [element]({{< relref "/how-to/bpmn/bpmn-elements" >}}) that has an `Output` or `Input response` parameter.
+It can use any data from the {{< abbr "process variable context" >}}, including variables added on the fly.
+Its format can use JSON or JSONata.
+
+Functionally, only the last value of the `customResponse` is returned to the client that sent the response.
+However, you can use conditional branches and different end nodes to add error handling.
+
+For example, this workflow returns `Workflow ran correctly` if the call variables include the message `CORRECT` and an error message in all other cases.
+
+<!-- vale off -->
+{{< bigFigure
+src="/images/bpmn/screenshot-rhize-bpmn-error-handling-custom-response.png"
+alt="A BPMN workflow with customResponse in the output of the end node"
+caption="Download this workflow from [BPMN templates](https://github.com/libremfg/bpmn-templates/)"
+width="80%"
+>}}
+<!-- vale on -->
+
+### Additional properties for workflow calls {#variables-versions}
+
+API operations can also include parameters to add variables to the {{< abbr "process variable context" >}} and to specify a workflow version.
+
+To add variables, use the `variables` input argument.
+Note that the variables are accessible from their root object.
+For example, a workflow would access the following string value at `$.input.message`:
+
+```gql{
+"variables": "{\"input\":{\"message\":\"CORRECT\"}}"
+}
+```
+
+To specify a version, use the `version` property. For example, this input specifies that Rhize run version `3` of the `API_demoCallToRemoteAPI` workflow:
+
+```json
+{
+  "createAndRunBpmnSyncId": "API_demoCallToRemoteAPI",
+  "version": "3"
+}
+```
+
+## Start from a message
+
+The [message start event]({{< relref "/how-to/bpmn/bpmn-elements#message-start-event" >}}) subscribes to a topic on the Rhize broker.
+Whenever a message is published to this topic, the workflow is triggered.
+The Rhize broker can receive messages published over MQTT, NATS, and OPC UA.
+
+For example, this workflow subscribes to the topic `material/stuff`.
+Whenever a message is published to the topic, it evaluates whether the quantity is in the correct threshold.
+If the quantity is correct, it uses the [mutation service task]({{< relref "/how-to/bpmn/bpmn-elements#graphql-mutation/" >}}) to add a material lot.
+If incorrect, it sends an alert back to the broker.
+
+<!-- vale off -->
+{{% bigFigure 
+alt="BPMN message start with conditional evaluation"
+src="/images/bpmn/rhize-bpmn-message-start-throw-conditional.png"
+width="65%"
+caption="Download this workflow as a [BPMN template](https://github.com/libremfg/bpmn-templates/tree/main/msg-start-and-throw)."
+ %}}
+<!-- vale on -->
+
+Note that, for a workflow to run from a message start event, the workflow **must be enabled.**
```

Comment:
Not sure if you want to go to this level of detail but there are two things:
1. The workflow needs an active version.
2. The WorkflowMessage of that active version needs to be enabled.

Reply:
I just checked and I think the UI does not permit the User to enable a workflow if it's not active, so for now I'm going to leave that part out. Of course we can add it back in if users complain.